persistence/data_models_enums.go

```diff
@@ -48,13 +48,14 @@ func (e ProcessExecutionStatus) String() string {
 type StateExecutionStatus int32
 
 const (
-	StateExecutionStatusSkipped   StateExecutionStatus = -1
-	StateExecutionStatusUndefined StateExecutionStatus = 0
-	StateExecutionStatusRunning   StateExecutionStatus = 1
-	StateExecutionStatusCompleted StateExecutionStatus = 2
-	StateExecutionStatusFailed    StateExecutionStatus = 3
-	StateExecutionStatusTimeout   StateExecutionStatus = 4
-	StateExecutionStatusAborted   StateExecutionStatus = 5
+	StateExecutionStatusSkipped            StateExecutionStatus = -1
+	StateExecutionStatusUndefined          StateExecutionStatus = 0
+	StateExecutionStatusRunning            StateExecutionStatus = 1
+	StateExecutionStatusCompleted          StateExecutionStatus = 2
+	StateExecutionStatusFailed             StateExecutionStatus = 3
+	StateExecutionStatusTimeout            StateExecutionStatus = 4
+	StateExecutionStatusAborted            StateExecutionStatus = 5
+	StateExecutionStatusWaitingForCommands StateExecutionStatus = 6
```

Comment:
Need a new status to prevent repeatedly running wait_until tasks that haven't been deleted.

Reply:
Having a new status makes sense. 

Question: is this before or after running status?

Reply:
Figured it out! it's after. So running->waiting->completed